staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -56,9 +56,12 @@ var defaultSleepDuration = 10 * time.Second
 
 // TODO: validate impact of copying and investigate hashing
 type eventKey struct {
+	eventType           string
 	action              string
 	reason              string
+	note                string
 	reportingController string
+	reportingInstance   string
```

Comment:
@wojtek-t any idea why this wasn't kept?  It seems to me that the instance reporting the failure is pertinent to deciding whether an event is the same.

Reply:
This is how reportingInstance is initialized:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/event_broadcaster.go#L164

So basically, it's used to distinguish different instances of the controllers, but it's e.g. not distinguishing between incarnations of the controller on the machine (e.g. across kubelet restarts).
While we don't want to distinguish the latter, we want to to distinguish the former.

So yes - this seems like a bug to me too. It should be included.